### DISCUSSION TITLE
Estou aprendendo.

### DISCUSSION THREAD

**Discussion by ghost** [frontendbr/forum#925] [Discussão]
Bem, já tentei várias vezes aprender desenvolvimento Front-End e das vídeos aulas que vi não gostei da didática do professor. Já assinei o Alura e não gostei por ter achado muito superficial (Gosto de me aprofundar em tudo). Até que então eu achei o curso do Andre Fontenelle de desenvolvedor web no udemy na promoção e estou gostando bastante. Quais boas práticas pra iniciantes vocês recomendariam? Já pesquisei no Google mas eu fico muito na duvida de como praticar e que tipos de projetos posso colocar em prática.  O que me recomendariam? Gostaria de ouvir de quem já está na área a um bom tempo e de quem está começando também.  (:

**Comment by luizbills** [comment]
- Caso você queria ser um *front-ender* mais programador recomendo que **procure um problema e desenvolva uma solução para ele**. Pode ser problema de outra pessoa, grupo, empresa ou de preferência um seu mesmo. 

- Se você quer ser um front-end mais na pegada designer, eu sugiro você fazer **copywork**. Que é nada mais que pegar um site/interface e recria-lo.

Repita várias vezes cada um desses tópicos e procure freelas também para te dar uma motivação ($) a mais.

**Comment by Leocardoso94** [comment]
Olha esse repositório [frontend-challenges](https://github.com/LFeh/frontend-challenges) tem diversos desafios para vagas de front-end. 

Creio que seria de grande valia tentar resolver alguns desses desafios com os conhecimentos que você adquiri no seus estudos.

Uma plataforma que me ajudou bastante com front-end no início foi o [freeCodeCamp](https://www.freecodecamp.org/) lá eu aprendi bastante coisa e tem vários desafios muito legais, e ainda te dá um certificado ao final dos cursos, tudo de graça!

**Comment by rubenmarcus** [comment]
Aprende inglês, porquê portugues tem muito cursos bons, mas os melhores e a maioria do conteúdo ta em inglês
e já que você falou no Udemy, compra o desses dois caras
[Maximilian Schwarzmüller](https://www.udemy.com/user/maximilian-schwarzmuller/)
[Stephen Grider](https://www.udemy.com/user/sgslo/)

Eu comprei todos de Angular, React e Vue e também Webpack,PWA e JS deles, e to querendo comprar o de ethereum e blockchain, a didatica é sensacional e eles não ficam só focados ou lendo documentação, eles explicam todo o contexto, muitas vezes mostrando comparações, falam sobre ES6, o porquê de se usar a tecnologia e etc.

Um app muito bom mesmo, instala no seu android ou ios , se chama **Enki**. Ele te propõe um treino  diário de vários temas entre eles Web e JS, e te ensina por exemplos, padrões de código, arquiteturas, situações, algoritmos, e depois faz um teste com você.

Outra coisa

Se cadastra no [**HackerRank**](https://hackerrank.com), e no **[Upwork](https://upwork.com)**, e como falou o @Leocardoso94  tem o front-end challenges. Fazer esses testes, faz você raciocinar melhor e aprender muito mais que cursos!! 

E por último, ajude em projetos **opensource**, e responda questões aqui ou no **Stack Overflow**. Traduza material no **Mozilla Docs**.. 

São algumas coisas que eu to fazendo, como Front hoje pra recuperar o tempo,e sempre se tem a sensação de ser junior, mas é uma sensação boa, porquê você sempre aprende mais,  e sente sua evolução acontecendo em pouco tempo..

**Comment by danielschmitz** [comment]
Você nao vai ficar bom apenas vendo vídeos. Aquilo ali é para dar um norte. Somente treinando você irá melhorar.

Recomendo:
https://github.com/LFeh/frontend-challenge

**Comment by ghost** [comment]
Então. Respondendo a pergunta do  @luizbills  quero ser mais _**programador**_. Eu peguei o básico hoje, fiz 6 seções do curso se não me engano e acabei fazendo uma lista do que quero estudar e o que estudar como prática básica. Tenho que implementar algumas coisas fora os exercícios do curso. Comecei o curso hoje e to gostando bastante.  E desculpe-me pela pergunta meio fora ou dentro da issue, cursos online como os do udemy é válido no mercado de trabalho?  E uma pergunta meio " fora" ou  não, como organizar meus repositórios? Eu tinha uma conta antiga no GitHub acabei excluindo pela desorganização que estava e não tinha/não tenho tanto conhecimento em Git/GitHub. Só sei abrir algumas issues e fazer pesquisa. É recomendável colocar TUDO o que eu fizer no GitHub?
